lib/templatesimplifier.cpp

```diff
@@ -521,7 +521,7 @@ unsigned int TemplateSimplifier::templateParameters(const Token *tok)
             return 0;
 
         // inner template
-        if (tok->str() == "<") {
+        if (tok->str() == "<" && !tok->previous()->isNumber()) {
```

Comment:
It seems it would be better to just do `tok->previous()->isName()` as that is the only token valid to precede a template argument.